tools/sync.js

```diff
@@ -0,0 +1,214 @@
+/*
```

Comment:
In summary I really like the enthusiasm and the desire to be fancy but I think it may be better if we're less fancy and more straightforward about how the syncing happens.